examples/PowerFlow/MatPowerTesting/MatPowerTesting.hpp

```diff
@@ -0,0 +1,219 @@
+    {
+      int fail = 0;
+
+      fail += a.bus_i != b.bus_i;
+      fail += a.type != b.type;
+      fail += a.Gs != b.Gs;
+      fail += a.Bs != b.Bs;
+      fail += a.area != b.area;
+      fail += !isEqual(a.Vm, b.Vm, tol);
+      fail += !isEqual(a.Va, b.Va, tol);
+      fail += a.baseKV != b.baseKV;
+      fail += a.zone != b.zone;
+      fail += !isEqual(a.Vmax, b.Vmax, tol);
+      fail += !isEqual(a.Vmin, b.Vmin, tol);
+
+      if (fail)
+      {
+        errs() << "bus_i: a=" << a.bus_i << ", b=" << b.bus_i << "\n"
+               << "type: a=" << a.type << ", b=" << b.type << "\n"
+               << "Gs: a=" << a.Gs << ", b=" << b.Gs << "\n"
+               << "Bs: a=" << a.Bs << ", b=" << b.Bs << "\n"
+               << "area: a=" << a.area << ", b=" << b.area << "\n"
+               << "Vm: a=" << a.Vm << ", b=" << b.Vm << "\n"
+               << "Va: a=" << a.Va << ", b=" << b.Va << "\n"
+               << "baseKV: a=" << a.baseKV << ", b=" << b.baseKV << "\n"
+               << "zone: a=" << a.zone << ", b=" << b.zone << "\n"
+               << "Vmax: a=" << a.Vmax << ", b=" << b.Vmax << "\n"
+               << "Vmin: a=" << a.Vmin << ", b=" << b.Vmin << "\n";
+      }
+      return fail == 0;
+    }
+
+    template <typename RealT = double, typename IdxT = int>
+    inline bool isEqual(PowerSystemData::LoadData<RealT, IdxT> a,
+                        PowerSystemData::LoadData<RealT, IdxT> b,
+                        RealT                                  tol = tol_)
+    {
+      int fail = 0;
+
+      fail += a.bus_i != b.bus_i;
+      fail += !isEqual(a.Pd, b.Pd, tol);
+      fail += !isEqual(a.Qd, b.Qd, tol);
+
+      if (fail)
+      {
+        errs() << "bus_i: a=" << a.bus_i << ", b=" << b.bus_i << "\n"
+               << "Pd: a=" << a.Pd << ", b=" << b.Pd << "\n"
+               << "Qd: a=" << a.Qd << ", b=" << b.Qd << "\n";
+      }
+      return fail == 0;
+    }
+
+    template <typename RealT = double, typename IdxT = int>
+    inline bool isEqual(PowerSystemData::BranchData<RealT, IdxT> a,
+                        PowerSystemData::BranchData<RealT, IdxT> b,
+                        RealT                                    tol = tol_)
+    {
+      int fail = 0;
+
+      fail += a.fbus != b.fbus;
+      fail += a.tbus != b.tbus;
+      fail += !isEqual(a.r, b.r, tol);
+      fail += !isEqual(a.x, b.x, tol);
+      fail += !isEqual(a.b, b.b, tol);
+      fail += a.rateA != b.rateA;
+      fail += a.rateB != b.rateB;
+      fail += a.rateC != b.rateC;
+      fail += a.ratio != b.ratio;
+      fail += a.angle != b.angle;
+      fail += a.status != b.status;
+      fail += a.angmin != b.angmin;
+      fail += a.angmax != b.angmax;
+
+      if (fail)
+      {
+        errs() << "Got failure!\na=" << a.str() << "\nb=" << b.str();
+      }
+      return fail == 0;
+    }
+
+    template <typename T>
+    inline bool isEqual(std::vector<T> a, std::vector<T> b, double tol = tol_)
```

Comment:
I would leave the comparison of vectors in `Utilities/Testing.hpp` and have this header file only operate on `PowerSystemData`.

Reply:
I tried that first, but this function template is not well defined and would require a little more work to fix. So far, only Matpower testing is using it. We can implement a separate function (template) for comparing vectors when/if needed.